plan/expression_rewriter.go

```diff
@@ -281,46 +243,139 @@ func (er *expressionRewriter) Leave(inNode ast.Node) (retNode ast.Node, ok bool)
-			return retNode, false
+	case *ast.IsTruthExpr:
+		op := ast.IsTruth
+		if v.True == 0 {
+			op = ast.IsFalsity
 		}
+		function := er.notToScalarFunc(v.Not, op, v.Type, er.ctxStack[stkLen-1])
+		er.ctxStack = er.ctxStack[:stkLen-1]
+		er.ctxStack = append(er.ctxStack, function)
+	case *ast.BinaryOperationExpr:
 		var function expression.Expression
 		switch v.Op {
 		case opcode.EQ, opcode.NE, opcode.NullEQ:
 			var err error
-			function, err = constructBinaryOpFunction(er.ctxStack[stkLen-2], er.ctxStack[stkLen-1], funcName)
+			function, err = constructBinaryOpFunction(er.ctxStack[stkLen-2], er.ctxStack[stkLen-1],
+				opcode.Ops[v.Op])
 			if err != nil {
 				er.err = errors.Trace(err)
 				return retNode, false
 			}
 		default:
-			function = expression.NewFunction(funcName, v.Type, er.ctxStack[stkLen-2], er.ctxStack[stkLen-1])
+			function = expression.NewFunction(opcode.Ops[v.Op], v.Type, er.ctxStack[stkLen-2], er.ctxStack[stkLen-1])
 		}
 		er.ctxStack = er.ctxStack[:stkLen-2]
 		er.ctxStack = append(er.ctxStack, function)
+	case *ast.BetweenExpr:
+		er.betweenToScalarFunc(v)
+	case *ast.PatternLikeExpr:
+		er.likeToScalarFunc(v)
+	case *ast.PatternInExpr:
+		er.inToScalarFunc(v)
 	case *ast.UnaryOperationExpr:
-		funcName, ok := opcode.Ops[v.Op]
-		if !ok {
-			er.err = errors.Errorf("Unknown opcode %v", v.Op)
-			return retNode, false
-		}
 		if getRowLen(er.ctxStack[stkLen-1]) != 1 {
 			er.err = errors.New("Operand should contain 1 column(s)")
 			return retNode, false
 		}
-		function := expression.NewFunction(funcName, v.Type, er.ctxStack[stkLen-1])
+		function := expression.NewFunction(opcode.Ops[v.Op], v.Type, er.ctxStack[stkLen-1])
 		er.ctxStack = er.ctxStack[:stkLen-1]
 		er.ctxStack = append(er.ctxStack, function)
-
 	default:
 		er.err = errors.Errorf("UnknownType: %T", v)
 		return retNode, false
 	}
+
+	if er.err != nil {
+		return retNode, false
+	}
 	return inNode, true
 }
+
+func (er *expressionRewriter) notToScalarFunc(b bool, op string, tp *types.FieldType,
+	args ...expression.Expression) *expression.ScalarFunction {
+	opFunc := expression.NewFunction(op, tp, args...)
+	v := int64(1)
+	if !b {
+		v = int64(0)
+	}
+	return expression.NewFunction(ast.UnaryNot, tp, intToExprConstant(v), opFunc)
+}
+
+func (er *expressionRewriter) inToScalarFunc(v *ast.PatternInExpr) {
+	stkLen := len(er.ctxStack)
+	lLen := len(v.List)
+	function := er.notToScalarFunc(v.Not, ast.In, v.Type, er.ctxStack[stkLen-lLen-1:stkLen]...)
+	er.ctxStack = er.ctxStack[:stkLen-lLen-1]
+	er.ctxStack = append(er.ctxStack, function)
+}
+
+func (er *expressionRewriter) likeToScalarFunc(v *ast.PatternLikeExpr) {
+	l := len(er.ctxStack)
+	function := er.notToScalarFunc(v.Not, ast.Like, v.Type,
+		er.ctxStack[l-2], er.ctxStack[l-1], intToExprConstant(int64(v.Escape)))
+	er.ctxStack = er.ctxStack[:l-2]
+	er.ctxStack = append(er.ctxStack, function)
+}
+
+func (er *expressionRewriter) betweenToScalarFunc(v *ast.BetweenExpr) {
```

Comment:
between function should be rewrited here. between(a,b,c) should be rewrited to andand(ge(a,b),le(a,c))